static/app/utils/oxfordizeArray.tsx

```diff
@@ -0,0 +1,38 @@
+import {Fragment} from 'react';
+
+// Given a list of strings (probably nouns), join them into a single string
+// with correct punctuation and 'and' placement
+//
+// for example: ['A'] --> 'A'
+//              ['A', 'B'] --> 'A and B'
+//              ['A', 'B', 'C'] --> 'A, B, and C'
+const oxfordizeArray = (strings: string[]) =>
+  strings.length <= 2
+    ? strings.join(' and ')
+    : [strings.slice(0, -1).join(', '), strings.slice(-1)[0]].join(', and ');
+
+export const oxfordizeElements = (elements: JSX.Element[]): JSX.Element => {
+  if (elements.length === 0) {
```

Comment:
It feels to me like this should just be a React Component with props {children: React.ReactNode} and we can then use React.Children(children).toArray() to perform the oxfordize logic. That way we have a declarative API for React components + it's "functional" counterpart for string[] types. Wdyt?

Reply:
Updated to use a component with children.